pymatgen/io/lammps/inputs.py

```diff
@@ -33,10 +33,114 @@
 class LammpsInputFile(InputFile):
     """
     Class representing a LAMMPS input settings file, e.g. in.lammps.
+    Allows for LAMMPS input generation in line/stage wise manner. A stage
+    here is defined as a block of LAMMPS input commands usually performing a
+    specific task during the simulation such as energy minimization or
+    NPT/NVT runs. But more broadly, a stage can also be block of LAMMPS
+    input where simulation box is setup, a set of variables are declared or
+    quantities are computed. A comment beginning with '#' is treated as a
+    header for a stage and marks the start of a new stage in the LAMMPS input.
+    Other comments starting with '##' are treated as conventional comments.
     """
 
     def __init__(self, input_settings: OrderedDict = None):
-        self.input_settings = input_settings
+        """
+        Args:
+            input_settings: Ordered Dictionary of LAMMPS input settings.
+        """
+
+        self.nstages = 0
+        self.ncomments = 0
+        self.input_settings = input_settings if input_settings is not None else OrderedDict()
+        self.curr_stage_name = None
+
+    def init_stage(self):
+        """
+        Use this to initiate new stage/black in LAMMPS input file.
+        """
+
+        self.nstages += 1
+
+        stage_name = "stage %s" % self.nstages
```

Comment:
Minor comment - here you use old style format strings but at line 84 you use an f-string. It's OK as-is but I'd suggest you keep your code style consistent throughout (and use f-strings everywhere since they are more modern)